core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -334,6 +334,17 @@ class SparkHadoopUtil extends Logging {
    * Stop the thread that does the delegation token updates.
    */
   private[spark] def stopExecutorDelegationTokenRenewer() {}
+
+  /**
+   * Disable the hadoop fs cache mechanism, otherwise DFSClient will use old token to connect nn.
```

Comment:
can you be more explicit here: `otherwise DFSClient will use old token to connect to the NameNode`. This piece of code is confusing enough. We should at least have clear comments.